cwms-data-api/src/test/java/cwms/cda/api/EntityControllerTestIT.java

```diff
@@ -0,0 +1,294 @@
+            .header(AUTH_HEADER, TestAccounts.KeyUser.SPK_NORMAL.toHeaderValue())
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .delete("/entity/" + ENTITY_ID)
+        .then()
+            .statusCode(isOneOf(
+                    HttpServletResponse.SC_NO_CONTENT,
+                    HttpServletResponse.SC_NOT_FOUND,
+                    HttpServletResponse.SC_BAD_REQUEST
+            ));
+    }
+
+    // Test CRUD  
+    // create -> getOne -> update -> getOne to verify updated
+    // delete -> getOne to verify deleted
+    @ParameterizedTest
+    @ValueSource(strings = {Formats.JSONV2, Formats.DEFAULT})
+    void test_entity_create_get_update_delete(String format) throws Exception {
+
+        TestAccounts.KeyUser user = TestAccounts.KeyUser.SPK_NORMAL;
+        InputStream in = this.getClass().getResourceAsStream("/cwms/cda/data/dto/entity.json");
+        Assertions.assertNotNull(in);
+        String entityJson = IOUtils.toString(in, java.nio.charset.StandardCharsets.UTF_8);
+
+        // CREATE
+        given()
+            .log().ifValidationFails(LogDetail.ALL, true)
+            .contentType(Formats.JSONV2)
+            .body(entityJson)
+            .header(AUTH_HEADER, user.toHeaderValue())
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .post("/entity")
+        .then()
+            .log().ifValidationFails(LogDetail.ALL, true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_CREATED));
+
+        // GET
+        given()
+            .log().ifValidationFails(LogDetail.ALL, true)
+            .accept(format)
+            .queryParam(Controllers.OFFICE, OFFICE_ID)
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .get("/entity/" + ENTITY_ID)
+        .then()
+            .log().ifValidationFails(LogDetail.ALL, true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_OK))
+            .body("id.name", equalTo(ENTITY_ID))
+            .body("id.office-id", equalTo(OFFICE_ID))
+            .body("long-name", equalTo("National Weather Service"));
+
+        // UPDATE — modify long-name to verify persistence
+        String updatedEntityJson = entityJson.replace(
+                "\"National Weather Service\"",
+                "\"National Weather Service (Updated)\"");
+
+        given()
+            .contentType(Formats.JSONV2)
+            .body(updatedEntityJson)
+            .header(AUTH_HEADER, user.toHeaderValue())
+            .queryParam(Controllers.OFFICE, OFFICE_ID)
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .patch("/entity/" + ENTITY_ID)
+        .then()
+            .log().ifValidationFails(LogDetail.ALL,true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_OK));
+
+        // GET to confirm updated field persisted
```

Comment:
Add an update-with-non-existing-entity-id check here with a bogus entity id to make sure its returning the not-found status.